ocaml/xapi/storage_smapiv1_wrapper.ml

```diff
@@ -603,8 +603,8 @@ functor
         match failures with [] -> next () | f :: _ -> raise f
 
       let epoch_begin context ~dbg ~sr ~vdi ~vm ~persistent =
-        with_dbg ~name:"VDI.epoch_begin" ~dbg @@ fun {log= dbg; _} ->
-        info "VDI.epoch_begin dbg:%s sr:%s vdi:%s vm:%s persistent:%b" dbg
+        with_dbg ~name:"VDI.epoch_begin" ~dbg @@ fun {log; _} ->
```

Comment:
Maybe I'm reading this wrong, but this change seems to contradict the commit message. This looks like a simple rename to me, but on the commit message it state that the type being passed around changes, same with the other functions.

What am I missing?

Reply:
My understanding was that 'with_dbg' transforms 'dbg' into a 'Debuginfo.t' record `{log:dbg_log; tracing: dbg_tracing}`. Therefore, 'dbg' is shadowed in the scope of 'f' with 'dbg_log' and the 'traceparent' is lost for 'Impl' functions.